lazy_theta_star_planner/src/lazy_theta_star2.cpp

```diff
@@ -0,0 +1,257 @@
+{
+  std::vector<coords<world_pts>> path_rev;
+  do {
+    coords<world_pts> world;
+    costmap_->mapToWorld(data[curr_id].x, data[curr_id].y, world.x, world.y);
+    path_rev.push_back({world.x, world.y});
+    curr_id = data[curr_id].parent_id;
+  } while (curr_id != 0);
+  coords<world_pts> w;
+  costmap_->mapToWorld(data[curr_id].x, data[curr_id].y, w.x, w.y);
+  path_rev.push_back({w.x, w.y});
+
+  for (int i = path_rev.size() - 1; i >= 0; i--) {
+    raw_points.push_back(path_rev[i]);
+  }
+}
+
+void LazyThetaStar::resetParent(tree_node & curr_data)
+{
+  map_pts mx, my;
+  id m_id;
+  cost min_dist = INF_COST;
+  id min_dist_id;
+
+  for (int i = 0; i < how_many_corners; i++) {
+    mx = curr_data.x + moves[i][0];
+    my = curr_data.y + moves[i][1];
+
+    if (withinLimits(mx, my)) {
+      getIndex(mx, my, m_id);
+      if (m_id != -1) {
+        if (data[m_id].f < min_dist) {
+          min_dist = data[m_id].f;
+          min_dist_id = m_id;
+        }
+      }
+    }
+  }
+  curr_data.parent_id = min_dist_id;
+  curr_data.g = data[min_dist_id].g +
+    dist(curr_data.x, curr_data.y, data[min_dist_id].x, data[min_dist_id].y);
+  curr_data.f = curr_data.g + curr_data.h;
+}
+
+bool LazyThetaStar::isGoal(map_pts & cx, map_pts & cy)
+{
+  return cx == dst.x && cy == dst.y;
+}
+void LazyThetaStar::setNeighbors(tree_node & curr_data)
+{
+
+  tree_node & curr_par = data[curr_data.parent_id];
+
+  map_pts mx, my;
+  id m_id;
+  for (int i = 0; i < how_many_corners; i++) {
+    mx = curr_data.x + moves[i][0];
+    my = curr_data.y + moves[i][1];
+
+    if (mx == curr_par.x && my == curr_par.y) {
+      continue;
+    }
+
+    if (withinLimits(mx, my)) {
+      cost g_cost = curr_par.g + dist(mx, my, curr_par.x, curr_par.y);
+      cost h_cost, cal_cost;
+      getIndex(mx, my, m_id);
+
+      if (isSafe(mx, my) && m_id == -1) {
+        h_cost = dist(mx, my, dst.x, dst.y);
+        cal_cost = g_cost + h_cost;
+        data.push_back({mx, my, g_cost, h_cost, curr_data.parent_id, -1, cal_cost});
+        pq.push({id_gen, &(data[id_gen].f)});
+        addIndex(mx, my, id_gen);
+        id_gen++;
+        continue;
+      } else if (m_id != -1) {
+        tree_node & curr_node = data[m_id];
+        h_cost = data[m_id].h;
+        cal_cost = g_cost + h_cost;
+        if (curr_node.f > cal_cost) {
+          curr_node.g = g_cost;
+          curr_node.f = cal_cost;
+          curr_node.parent_id = curr_data.parent_id;
+          if (curr_node.closed == 1) {
+            curr_node.closed = -1;
+            pq.push({m_id, &(data[m_id].f)});
+          }
+        }
+        continue;
+      }
+    }
+  }
+}
+
+void LazyThetaStar::initializeStuff()
+{
+  id_gen = 0;
+  if (costmap_->getCost(src.x,
+    src.y) > lethal_cost && costmap_->getCost(src.x, src.y) < LETHAL_COST)
+  {
+    lethal_cost = costmap_->getCost(src.x, src.y);
+  }
+  sizeX = costmap_->getSizeInCellsX();
```

Comment:
If these are member variables they should be named as such `sizeX_`, same for data, etc